Generated client code missing exceptions

Some of the stub code the Obsidian compiler generates does not compile. In particular, for the Insurance case study, the code compiles (with Obsidian), but has the following error when built with Gradle: 

```
> Task :compileJava FAILED
/home/roei/ObsidianImports/obs_output/Insurer/src/main/java/org/hyperledger/fabric/example/Money__Stub__.java:26: error: unreported exception InvalidProtocolBufferException; must be caught or declared to be thrown
        argArray.add(Base64 .getEncoder().encodeToString(m.__wrappedArchiveBytes()));
                                                                                ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

--- a/resources/tests/compilerTests/StubExceptions.obs
+++ b/resources/tests/compilerTests/StubExceptions.obs
@@ -1,0 +1,4 @@
+import "resources/tests/compilerTests/StubExceptionsLib.obs"
+
+main contract StubExceptions {}
+

--- a/resources/tests/compilerTests/StubExceptionsLib.obs
+++ b/resources/tests/compilerTests/StubExceptionsLib.obs
@@ -1,0 +1,8 @@
+contract C {
+    transaction f(C@Unowned other) {}
+
+    transaction g() {
+        f(new C());
+    }
+}
+

--- a/src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGen.scala
+++ b/src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGen.scala
@@ -380,8 +380,8 @@ class CodeGen (val target: Target, table: SymbolTable) {
         }
 
         val meth: JMethod = newClass.method(JMod.PUBLIC, javaRetType, transaction.name)
+        addTransactionExceptions(meth, translationContext)
         meth._throws(model.directClass("edu.cmu.cs.obsidian.client.ChaincodeClientAbortTransactionException"))
-
 
         var argExpressions: List[IJExpression] = Nil
         /* add args */

--- a/src/test/scala/edu/cmu/cs/obsidian/tests/CompilerTests.scala
+++ b/src/test/scala/edu/cmu/cs/obsidian/tests/CompilerTests.scala
@@ -9,10 +9,9 @@ import org.scalatest.junit.JUnitSuite
 import scala.sys.process._
 
 class CompilerTests extends JUnitSuite {
-
-  def testContract(contractName : String) = {
+  def testContract(contractName: String, contractPath: String, args: Array[String]): Int = {
     var result = true
-    val inputArgs: Array[String] = Array(s"--output-path", s"obs_output/", s"resources/tests/compilerTests/$contractName.obs")
+    val inputArgs: Array[String] = Array(s"--output-path", s"obs_output/", contractPath) ++ args
     result = Main.compileProgram(inputArgs)
     assertTrue(result)
     val gradleCmd = s"gradle compileJava -b obs_output/$contractName/build.gradle"
@@ -21,6 +20,9 @@ class CompilerTests extends JUnitSuite {
     val deleteCmd = s"rm -rf obs_output/$contractName"
     deleteCmd.!
   }
+
+  def testContract(contractName: String): Int =
+    testContract(contractName, s"resources/tests/compilerTests/$contractName.obs", Array())
 
   @Test def multipleConstructors(): Unit = {
       testContract("MultipleConstructors")
@@ -72,5 +74,13 @@ class CompilerTests extends JUnitSuite {
 
   @Test def multipleConstructorGroup(): Unit = {
     testContract(contractName = "MultipleConstructorGroup")
+  }
+
+  @Test def insuranceCaseStudy(): Unit = {
+    testContract("Insurer", "resources/case_studies/Insurance/Insurer.obs", Array())
+  }
+
+  @Test def stubExceptions(): Unit = {
+    testContract("StubExceptions")
   }
 }

